vernac/comExtraDeps.ml

```diff
@@ -11,7 +11,14 @@
 open Names
 open CErrors
 
-let rec ensure_only_one_path_contains from file acc = function
+let warn_file_found_multiple_times =
+  CWarnings.create ~name:"ambiguous-extra-dep" ~category:"filesystem"
+    (fun (file,from,other,extra) ->
+      Pp.(str "File " ++ str file ++ str " found twice in " ++
+      DirPath.print from ++ str":" ++ spc () ++ str other ++ str " (selected)," ++
+      spc() ++ str extra ++ str ".") )
+
+let rec first_path_containing ?loc from file acc = function
```

Comment:
That went under my radar, but certain that this function doesn't belong here but on common code.

How is this code kept in sync with the one in coqdep?

Reply:
Hum I guess you want me to use some code in tools/coqdep/lib.
The code there picks the first entry as well:
https://github.com/coq/coq/blob/2514b59568391079d0f3c362f6112b67fc380998/tools/coqdep/lib/common.ml#L228-L233

I did a quick look at the mli files, and I'm sure there is an API for this. Is it?

Reply:
done